integration_tests/serviceprovider_mock.go

```diff
@@ -33,6 +33,11 @@ type TestServiceProvider struct {
 	GetOauthEndpointImpl  func() string
 }
 
+func (t TestServiceProvider) CheckRepositoryAccess(ctx context.Context, cl client.Client, accessCheck *api.SPIAccessCheck) (*api.SPIAccessCheckStatus, error) {
+	//TODO implement me
+	panic("implement me")
```

Comment:
Make this injectable as the other method impls of the `TestServiceProvider`?

Reply:
yes